File: app/src/main/java/org/ametro/ui/navigation/entities/NavigationTextItem.java
package org.ametro.ui.navigation.entities;

import android.graphics.drawable.Drawable;

public class NavigationTextItem extends NavigationItem {

    private final Drawable icon;
    private final CharSequence text;

    public NavigationTextItem(int action, Drawable icon, CharSequence text) {
        this(action, icon, text, true, null);
    }

    public NavigationTextItem(int action, Drawable icon, CharSequence text, boolean enabled, Object source) {
        super(action, enabled, source);
        this.icon = icon;
        this.text = text;
    }

    public CharSequence getText() {
        return text;
    }

    public Drawable getDrawable() {
        return icon;
    }
}

File: app/src/main/java/io/github/romangolovanov/apps/ametro/ui/navigation/entities/NavigationTextItem.kt
package io.github.romangolovanov.apps.ametro.ui.navigation.entities

import android.graphics.drawable.Drawable

class NavigationTextItem : NavigationItem {

    private val icon: Drawable?
    val text: CharSequence

    constructor(action: Int, icon: Drawable?, text: CharSequence) : this(action, icon, text, true, null)

    constructor(action: Int, icon: Drawable?, text: CharSequence, enabled: Boolean, source: Any?) : super(action, enabled, source) {
        this.icon = icon
        this.text = text
    }

    fun getDrawable(): Drawable? = icon
}
